frontend/public/components/app.jsx

```diff
@@ -136,82 +139,85 @@ class App extends React.PureComponent {
-          <Route path="/k8s/cluster/clusterroles/:name" component={props => <ResourceDetailsPage {...props} plural="clusterroles" />} />
-
-          <Route path="/k8s/ns/:ns/roles/:name/add-rule" exact component={EditRulePage} />
-          <Route path="/k8s/ns/:ns/roles/:name/:rule/edit" exact component={EditRulePage} />
-          <Route path="/k8s/ns/:ns/roles" exact component={rolesListPage} />
-
-          <Route path="/k8s/ns/:ns/secrets/new/:type" exact component={props => <CreateSecret {...props} kind="Secret" />} />
-          <Route path="/k8s/ns/:ns/secrets/:name/edit" exact component={props => <EditSecret {...props} kind="Secret" />} />
-          <Route path="/k8s/ns/:ns/secrets/:name/edit-yaml" exact component={props => <EditYAMLPage {...props} kind="Secret" />} />
-
-          <Route path="/k8s/cluster/rolebindings/new" exact component={props => <CreateRoleBinding {...props} kind="RoleBinding" />} />
-          <Route path="/k8s/ns/:ns/rolebindings/new" exact component={props => <CreateRoleBinding {...props} kind="RoleBinding" />} />
-          <Route path="/k8s/ns/:ns/rolebindings/:name/copy" exact component={props => <CopyRoleBinding {...props} kind="RoleBinding" />} />
-          <Route path="/k8s/ns/:ns/rolebindings/:name/edit" exact component={props => <EditRoleBinding {...props} kind="RoleBinding" />} />
-          <Route path="/k8s/cluster/clusterrolebindings/:name/copy" exact component={props => <CopyRoleBinding {...props} kind="ClusterRoleBinding" />} />
-          <Route path="/k8s/cluster/clusterrolebindings/:name/edit" exact component={props => <EditRoleBinding {...props} kind="ClusterRoleBinding" />} />
-
-          <Route path="/k8s/cluster/:plural" exact component={ResourceListPage} />
-          <Route path="/k8s/cluster/:plural/new" exact component={CreateYAML} />
-          <Route path="/k8s/cluster/:plural/:name" component={ResourceDetailsPage} />
-          <Route path="/k8s/ns/:ns/pods/:podName/containers/:name" component={ContainersDetailsPage} />
-          <Route path="/k8s/ns/:ns/:plural/new" exact component={NamespaceFromURL(CreateYAML)} />
-          <Route path="/k8s/ns/:ns/:plural/:name" component={ResourceDetailsPage} />
-          <Route path="/k8s/ns/:ns/:plural" exact component={ResourceListPage} />
-
-          <Route path="/k8s/all-namespaces/:plural" exact component={ResourceListPage} />
-          <Route path="/k8s/all-namespaces/:plural/:name" component={ResourceDetailsPage} />
-
-          <Route path="/settings/profile" exact component={ProfilePage} />
-          <Route path="/settings/ldap" exact component={LDAPPage} />
-          <Route path="/settings/cluster" exact component={ClusterSettingsPage} />
-
-          <Route path="/error" exact component={ErrorPage} />
-          <Route path="/" exact component={DefaultPage} />
-
-          <Route component={ErrorPage404} />
-        </Switch>
-      </div>
-    </React.Fragment>;
+    return <DragDropContextProvider backend={HTML5Backend}>
```

Comment:
This will break async loading of the component, won't it?

Reply:
No, its not breaking the async loading. Tested it via the inspector on the name-value-editor and its loaded asynchronously, so this shouldn't effect it.

Reply:
But is react-dnd loaded async? That's the dependency we're worried about.

Since we have an impoty in app.jsx now, I don't think it can be.

Reply:
I thought that we want to import the react-dnd in app.jsx, so we ensure the HTML5Backend is loaded only once, for the whole console.
But I can rework it ina  way it will be loaded async only in the components that will use DnD.

Reply:
> I thought that we want to import the react-dnd in app.jsx, so we ensure the HTML5Backend is loaded only once, for the whole console.

Ideally we'd load react-dnd async since it's not small and only used in a couple places. If so, we should initialize the HTML5Backend once on first load.

The reason we made the name value editor async was to lazy load react-dnd.